### PR TITLE
Move code checks to Github Actions and enable staticcheck

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:
This makes sure to clean up more code that staticcheck flags as incorrect. My goal is to be able to run staticcheck on the whole codebase as part of CI, but it is currently held up by a few issues. Hopefully those can get ironed out in the future to give us some better code checks :slightly_smiling_face: 

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [ ] Public APIs match existing style.
- [ ] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

**Review by stuartmscott — APPROVED** [review]
Always good to see PRs that improve the code base

File: cmd/fyne_settings/settings/bundled.go [hunk 1]
```diff
@@ -16,7 +16,9 @@ var appearanceIcon = &fyne.StaticResource{
...
+*/
```
Comment (stuartmscott) [inline cmd/fyne_settings/settings/bundled.go:24]: See the first line in this file - this change will get reverted the next time the file is generated. Instead remove https://github.com/fyne-io/fyne/blob/master/cmd/fyne_settings/data/gen.go#L60

**Review by stuartmscott — CHANGES_REQUESTED** [review]

File: cmd/fyne_settings/settings/bundled.go (hunk 1 above)
Comment (Jacalz) [reply cmd/fyne_settings/settings/bundled.go:24]: Yeah good catch. I don't know why I didn't think of that :sweat_smile:

**Review by stuartmscott — APPROVED** [review]

**Review by stuartmscott — CHANGES_REQUESTED** [review]
Travis has a failure:
`fatal error: runtime: out of memory`

**Comment by Jacalz** [comment]
Hm, that is a weird one. I guess that `staticcheck` can be a bit memory hungry :(

**Comment by Jacalz** [comment]
Have been working on moving code checks out to Github Actions. I guess that now might be the time :question:

**Comment by dominikh** [comment]
> Hm, that is a weird one. I guess that staticcheck can be a bit memory hungry :(

@Jacalz Were you using the latest release (2020.1.x) of staticcheck, or the master branch? The master branch should use significantly less memory, and if it still OOM'd in Travis then that's probably something I need to investigate.

**Comment by Jacalz** [comment]
@dominikh I was using the latest stable release, I think. That should be what a regular `go get` should give us right?

I know that the stable releases have been quite memory hungry on my own machine (before I upgraded to 16GB of memory from 8GB, my Linux machine used to almost max out memory usage). I moved over to GitHub Actions to try to avoid Travis and that seemed to work fine.

**Comment by dominikh** [comment]
> That should be what a regular go get should give us right?

In module mode, yes. In GOPATH mode, that downloads the master branch.

I'll make a note to check staticcheck's memory usage on this project.

**Review by andydotxyz — APPROVED** [review]

**Review by stuartmscott — APPROVED** [review]

**Pull request merged**